syntax = "proto3";
package gw2l;
import "common.proto";
import "mysql_database_table.proto";
import "google/protobuf/empty.proto";
option cc_generic_services = true;



message LoginRequest
{
  string account = 1;
  string password = 2;
  uint64 conn_id = 3;
  uint32 gate_node_id = 4;
}

message LoginResponse
{
	ClientError error = 1;
  account_database account_player = 2;
}

message CreatePlayerRequest
{
  uint64 conn_id = 1;
}

message CreatePlayerResponse
{
  ClientError error = 1;
  account_database account_player = 2;
  uint64 guid = 3;
}

message EnterGameRequest
{
  uint64 conn_id = 1;
  uint64 guid = 2;
}

message EnterGameResponse
{
  ClientError error = 1;
  uint64 guid = 2;
  uint64 conn_id = 3;// for logic
}

message DisconnectRequest
{
  uint64 conn_id = 1;
}

message DisconnectResponse
{
}

message LeaveGameRequest
{
  uint64 conn_id = 1;
}

service LoginService {
  rpc Login (LoginRequest) returns (LoginResponse);
  rpc CreatPlayer (CreatePlayerRequest) returns (CreatePlayerResponse);
  rpc EnterGame (EnterGameRequest) returns (EnterGameResponse);
  rpc LeaveGame (LeaveGameRequest) returns (google.protobuf.Empty);
  rpc Disconnect (DisconnectRequest) returns (google.protobuf.Empty);
}
